src/routing/approved.rs

```diff
@@ -1399,25 +1384,17 @@ impl Approved {
             .ok_or(Error::InvalidSrcLocation)?
             .peer;
 
-        if !proofs.verify(&elders_info) {
-            error!(
-                "Ignore DKG failure agreement with invalid proofs: {}",
-                elders_info
-            );
-            return Ok(vec![]);
-        }
-
-        if !self
+        let elders_info = self
             .section
             .promote_and_demote_elders(&self.node.name())
-            .contains(&elders_info)
-        {
-            trace!(
-                "Ignore DKG failure agreement for outdated participants: {}",
-                elders_info
-            );
+            .into_iter()
+            .find(|elders_info| proofs.verify(elders_info, self.section.chain().len() as u64));
```

Comment:
Don't think that can happen because the current elders are all on the same key. But what can happen in theory is that some elders receive a `Sync` message which creates a fork somewhere further down the chain which still increases the length of the chain and causes them to use a different generation number than the other elders. Don't think it's very likely, but perhaps we should use the length of the main branch of the chain instead of the whole chain?

Reply:
> but perhaps we should use the length of the main branch of the chain instead of the whole chain?

related to the comment at https://github.com/maidsafe/sn_routing/pull/2336#discussion_r587301885
I think yes, we may need to think about this

It's not necessarily to be addressed within this PR (if you think the scenario is rare).
How about you raise it as an issue, so that we can tackle it later on ?